analyses/survival-analysis/survival-analysis_immune.Rmd

```diff
@@ -0,0 +1,266 @@
+---
+title: "Survival analysis by immune cells"
+output: html_notebook
+---
+
+This notebook will do the following multivariate survival analyses:
+1. Multivariate analysis using immune cell types (minus uncharacterized), PDL1 expression, and cancer group as covariates
+2. Multivariate analysis for each cancer group, using immune cell types (minus uncharacterized) and PDL1 expression as covariates
+3. Multivariate analysis for each cancer group, using immune cell types (minus uncharacterized), PDL1 expression, and molecular subtype as covariates
+
+## Setup
+
+#### Packages
+
+```{r Set up library}
+library(survival)
+library(dplyr)
+library(ggplot2)
+library(ggpubr)
+
+```
+
+#### Establish directories
+```{r}
+# define directories
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+input_dir <- file.path(root_dir, "analyses", "immune-deconv")
+analysis_dir <- file.path(root_dir, "analyses", "survival-analysis")
+source(file.path(analysis_dir, "util", "ggforest_new.R"))
+
+results_dir <- file.path(analysis_dir, "results")
+if(!dir.exists(results_dir)){
+  dir.create(results_dir)
+}
+
+plots_dir <- file.path(analysis_dir, "plots")
+if(!dir.exists(plots_dir)){
+  dir.create(plots_dir)
+}
+```
+
+#### Read in required files
+```{r}
+# Read in quantiseq immune deconvoluted scores file 
+quantiseq <- readRDS(file.path(input_dir, "results", "quantiseq_deconv-output.rds")) %>%
+  # remove uncharacterized
+  dplyr::filter(cell_type != "uncharacterized cell")
+
+# Read in gene expression files - both poly-A and stranded samples
+polya <- readRDS(file.path(data_dir, "pbta-gene-expression-rsem-fpkm-collapsed.polya.rds"))
+stranded <- readRDS(file.path(data_dir, "pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds"))
+
+# Read in histology file
+histology_df <- readr::read_tsv(file.path(data_dir, "pbta-histologies.tsv"))
+
+```
+
+#### filter to cancer group of interest 
+```{r}
+# Filter to all cancer groups of interest 
+all_cancer_groups_of_interest <- c("High-grade glioma astrocytoma", 
+                                   "Diffuse midline glioma",
+                                   "Low-grade glioma astrocytoma", 
+                                   "Ependymoma", 
+                                   "Medulloblastoma")
+
+# filter to cancer group of interest 
+histologies_df_filtered <- histology_df %>%
+  # RNA-Seq samples only
+  dplyr::filter(experimental_strategy == "RNA-Seq",
```

Comment:
Note that there is probably no need to use `dplyr::` if you load the `dplyr` library.